src/pages/resultsView/enrichments/MutationEnrichmentsTab.tsx

```diff
@@ -21,24 +22,27 @@ export default class MutationEnrichmentsTab extends React.Component<IMutationEnr
     private uniqueColorGetter = makeUniqueColorGetter();
 
     @autobind
-    private onProfileChange(molecularProfile: MolecularProfile) {
-        this.props.store._selectedEnrichmentMutationProfile = molecularProfile;
+    private onProfileChange(profileSet:{[id:string]:MolecularProfile}) {
+        this.props.store.setMutationEnrichmentProfileSet(profileSet);
     }
 
     public render() {
-        if (this.props.store.mutationEnrichmentData.isPending) {
+        if (this.props.store.studies.isPending || this.props.store.mutationEnrichmentData.isPending || this.props.store.selectedStudyMutationEnrichmentProfileSet.isPending) {
```

Comment:
you could use getMobxPromiseGroupStatus to simpify this.    or maybe @adamabeshouse  MobxView